### PR TITLE
storage/conformance: update conformance tests with multiple entries per resource

### PR DESCRIPTION
**Pull request by rithujohn191**
Fixes https://github.com/coreos/dex/issues/849.

### CODE REVIEW

**Comment by ericchiang** [comment]
@rithujohn191 I got an error running the kubernetes conformance tests.

```
$ go test -v github.com/coreos/dex/storage/kubernetes
=== RUN   TestIDToName
--- PASS: TestIDToName (0.00s)
=== RUN   TestOfflineTokenName
--- PASS: TestOfflineTokenName (0.00s)
=== RUN   TestNamespaceFromServiceAccountJWT
--- PASS: TestNamespaceFromServiceAccountJWT (0.00s)
=== RUN   TestLoadClient
time="2017-03-15T15:15:18-07:00" level=error msg="third party resource already created auth-code.oidc.coreos.com" 
time="2017-03-15T15:15:18-07:00" level=error msg="third party resource already created auth-request.oidc.coreos.com" 
time="2017-03-15T15:15:18-07:00" level=error msg="third party resource already created o-auth2-client.oidc.coreos.com" 
time="2017-03-15T15:15:18-07:00" level=error msg="third party resource already created signing-key.oidc.coreos.com" 
time="2017-03-15T15:15:18-07:00" level=error msg="third party resource already created refresh-token.oidc.coreos.com" 
time="2017-03-15T15:15:18-07:00" level=error msg="third party resource already created password.oidc.coreos.com" 
time="2017-03-15T15:15:18-07:00" level=error msg="third party resource already created offline-sessions.oidc.coreos.com" 
--- PASS: TestLoadClient (0.43s)
=== RUN   TestURLFor
--- PASS: TestURLFor (0.00s)
=== RUN   TestStorage
time="2017-03-15T15:15:18-07:00" level=error msg="third party resource already created auth-code.oidc.coreos.com" 
time="2017-03-15T15:15:18-07:00" level=error msg="third party resource already created auth-request.oidc.coreos.com" 
time="2017-03-15T15:15:18-07:00" level=error msg="third party resource already created o-auth2-client.oidc.coreos.com" 
time="2017-03-15T15:15:18-07:00" level=error msg="third party resource already created signing-key.oidc.coreos.com" 
time="2017-03-15T15:15:18-07:00" level=error msg="third party resource already created refresh-token.oidc.coreos.com" 
time="2017-03-15T15:15:18-07:00" level=error msg="third party resource already created password.oidc.coreos.com" 
time="2017-03-15T15:15:18-07:00" level=error msg="third party resource already created offline-sessions.oidc.coreos.com" 
=== RUN   TestStorage/AuthCodeCRUD
=== RUN   TestStorage/AuthRequestCRUD
=== RUN   TestStorage/ClientCRUD
=== RUN   TestStorage/RefreshTokenCRUD
=== RUN   TestStorage/PasswordCRUD
=== RUN   TestStorage/KeysCRUD
=== RUN   TestStorage/OfflineSessionCRUD
=== RUN   TestStorage/GarbageCollection
=== RUN   TestStorage/TimezoneSupport
=== RUN   TestStorage/AuthRequestConcurrentUpdate
=== RUN   TestStorage/ClientConcurrentUpdate
=== RUN   TestStorage/PasswordConcurrentUpdate
=== RUN   TestStorage/KeysConcurrentUpdate
--- FAIL: TestStorage (6.97s)
    --- PASS: TestStorage/AuthCodeCRUD (0.48s)
    --- PASS: TestStorage/AuthRequestCRUD (0.39s)
    --- PASS: TestStorage/ClientCRUD (0.64s)
    --- PASS: TestStorage/RefreshTokenCRUD (0.58s)
    --- PASS: TestStorage/PasswordCRUD (0.64s)
    --- PASS: TestStorage/KeysCRUD (0.43s)
    --- FAIL: TestStorage/OfflineSessionCRUD (0.31s)
    	conformance.go:491: create offline session: ID already exists
    --- PASS: TestStorage/GarbageCollection (1.11s)
    --- PASS: TestStorage/TimezoneSupport (0.26s)
    --- PASS: TestStorage/AuthRequestConcurrentUpdate (0.40s)
    --- PASS: TestStorage/ClientConcurrentUpdate (0.40s)
    --- PASS: TestStorage/PasswordConcurrentUpdate (0.39s)
    --- PASS: TestStorage/KeysConcurrentUpdate (0.57s)
FAIL
exit status 1
FAIL	github.com/coreos/dex/storage/kubernetes	7.399s
```

**Comment by rithujohn191** [comment]
[rjohn@localhost dex]$ go test -v ./storage/kubernetes
=== RUN   TestIDToName
--- PASS: TestIDToName (0.00s)
=== RUN   TestOfflineTokenName
--- PASS: TestOfflineTokenName (0.00s)
=== RUN   TestNamespaceFromServiceAccountJWT
--- PASS: TestNamespaceFromServiceAccountJWT (0.00s)
=== RUN   TestLoadClient
time="2017-03-16T11:35:59-07:00" level=info msg="third party resource already created auth-code.oidc.coreos.com" 
time="2017-03-16T11:36:00-07:00" level=info msg="third party resource already created auth-request.oidc.coreos.com" 
time="2017-03-16T11:36:00-07:00" level=info msg="third party resource already created o-auth2-client.oidc.coreos.com" 
time="2017-03-16T11:36:00-07:00" level=info msg="third party resource already created signing-key.oidc.coreos.com" 
time="2017-03-16T11:36:00-07:00" level=info msg="third party resource already created refresh-token.oidc.coreos.com" 
time="2017-03-16T11:36:00-07:00" level=info msg="third party resource already created password.oidc.coreos.com" 
time="2017-03-16T11:36:01-07:00" level=info msg="third party resource already created offline-sessions.oidc.coreos.com" 
--- PASS: TestLoadClient (2.81s)
=== RUN   TestURLFor
--- PASS: TestURLFor (0.00s)
=== RUN   TestStorage
time="2017-03-16T11:36:02-07:00" level=info msg="third party resource already created auth-code.oidc.coreos.com" 
time="2017-03-16T11:36:02-07:00" level=info msg="third party resource already created auth-request.oidc.coreos.com" 
time="2017-03-16T11:36:02-07:00" level=info msg="third party resource already created o-auth2-client.oidc.coreos.com" 
time="2017-03-16T11:36:02-07:00" level=info msg="third party resource already created signing-key.oidc.coreos.com" 
time="2017-03-16T11:36:02-07:00" level=info msg="third party resource already created refresh-token.oidc.coreos.com" 
time="2017-03-16T11:36:02-07:00" level=info msg="third party resource already created password.oidc.coreos.com" 
time="2017-03-16T11:36:03-07:00" level=info msg="third party resource already created offline-sessions.oidc.coreos.com" 
=== RUN   TestStorage/AuthCodeCRUD
=== RUN   TestStorage/AuthRequestCRUD
=== RUN   TestStorage/ClientCRUD
=== RUN   TestStorage/RefreshTokenCRUD
=== RUN   TestStorage/PasswordCRUD
=== RUN   TestStorage/KeysCRUD
=== RUN   TestStorage/OfflineSessionCRUD
=== RUN   TestStorage/GarbageCollection
=== RUN   TestStorage/TimezoneSupport
=== RUN   TestStorage/AuthRequestConcurrentUpdate
=== RUN   TestStorage/ClientConcurrentUpdate
=== RUN   TestStorage/PasswordConcurrentUpdate
=== RUN   TestStorage/KeysConcurrentUpdate
--- PASS: TestStorage (42.68s)
    --- PASS: TestStorage/AuthCodeCRUD (2.75s)
    --- PASS: TestStorage/AuthRequestCRUD (2.65s)
    --- PASS: TestStorage/ClientCRUD (4.03s)
    --- PASS: TestStorage/RefreshTokenCRUD (3.61s)
    --- PASS: TestStorage/PasswordCRUD (3.95s)
    --- PASS: TestStorage/KeysCRUD (2.25s)
    --- PASS: TestStorage/OfflineSessionCRUD (3.14s)
    --- PASS: TestStorage/GarbageCollection (6.10s)
    --- PASS: TestStorage/TimezoneSupport (1.60s)
    --- PASS: TestStorage/AuthRequestConcurrentUpdate (2.60s)
    --- PASS: TestStorage/ClientConcurrentUpdate (2.51s)
    --- PASS: TestStorage/PasswordConcurrentUpdate (2.22s)
    --- PASS: TestStorage/KeysConcurrentUpdate (3.18s)
PASS
ok  	github.com/coreos/dex/storage/kubernetes	45.499s

File: storage/conformance/conformance.go [hunk 1]
```diff
@@ -389,32 +486,46 @@ func testPasswordCRUD(t *testing.T, s storage.Storage) {
...
 	mustBeErrNotFound(t, "password", err)
 
 }
 
 func testOfflineSessionCRUD(t *testing.T, s storage.Storage) {
-	session := storage.OfflineSessions{
-		UserID:  "User",
-		ConnID:  "Conn",
+	session1 := storage.OfflineSessions{
+		UserID:  "User1",
```
Comment (ericchiang) [inline storage/conformance/conformance.go]: Please make this random so runs of tests aren't dependent on one another.

Comment (rithujohn191) [inline storage/conformance/conformance.go]: Done

**Comment by rithujohn191** [comment]
Fixed lint errors

**Pull request merged**